src/drivers/infiniband/hermon.c

```diff
@@ -3936,6 +3927,23 @@ static int hermon_probe ( struct pci_device *pci ) {
 	if ( ( rc = hermon_get_cap ( hermon ) ) != 0 )
 		goto err_get_cap;
 
+	for ( i = 0 ; i < hermon->cap.num_ports ; i++ ) {
+		port = &hermon->port[i];
+
+		/* Retrieve MAC address */
+		if ( ( rc = hermon_cmd_query_port ( hermon, HERMON_PORT_BASE + i,
+						    &query_port ) ) != 0 ) {
+			DBGC ( hermon, "Hermon %p port %d could not query port: %s\n",
+			       hermon, i, strerror ( rc ) );
+			goto err_query_port;
+		}
+
+		/* The MAC addresses are stored as consecutive bytes, so we have
+		 * to undo the implicit byte-swap in MLX_GET */
+		port->eth_mac.part.h = cpu_to_be16 ( MLX_GET ( &query_port, mac_47_32 ) );
+		port->eth_mac.part.l = cpu_to_be32 ( MLX_GET ( &query_port, mac_31_0 ) );
+	}
+
```

Comment:
Modified and merged as https://github.com/ipxe/ipxe/commit/699b9f1d1.  I realised that we could capture the MAC during `hermon_set_port_type()` and save a duplicate `hermon_cmd_query_port()`.  Also reworded the commit message to mention the potential breaking change for anyone who had DHCP reservations based on the old LEMAC.

Reply:
That commit link doesn't work for me, currently

Reply:
It does now - that looks like a good solution!